src/main/java/gregtech/api/util/GTUtility.java

```diff
@@ -920,4 +920,43 @@ public double getAsDouble() {
     public static int safeCastLongToInt(long v) {
         return v > Integer.MAX_VALUE ? Integer.MAX_VALUE : (int) v;
     }
+
+    /**
+     * Scales a proposed recipe voltage according to a provided Material working tier.
+     *
+     * @param voltage     The suggested base voltage for the recipe.
+     * @param workingTier The voltage tier ({@link GTValues#V}) to conform this recipe to.
+     *
+     * @return The new recipe voltage.
+     */
+    public static long scaleVoltage(long voltage, int workingTier) {
+        if (workingTier == GTValues.LV) {
+            // no action needed, as this is the default.
+            return voltage;
+        }
+
+        int tierLower = Math.max(0, workingTier - 1);
+        if (voltage > GTValues.VOC[tierLower]) {
+            // no action needed, this recipe is already scaled accordingly.
+            return voltage;
+        }
+
+        // Multiplying very low voltages (less than 16 EU/t) with a formula like below can cause undesirably high
+        // EU/t. For example, 2 EU/t scaled to MV would multiply all the way to 128 EU/t before being above LV voltage.
+        // For this reason, recipes below 16 EU/t simply get set to their tier's VHA[tier] (half voltage adjusted).
+        if (voltage <= 16) {
+            return GTValues.VHA[workingTier];
```

Comment:
This will index out of bounds with tiers > MAX. See my other comment disallowing >MAX entirely.